src/wallet/coin_selection.rs

```diff
@@ -1339,4 +1495,140 @@ mod test {
         assert!(result.selected_amount() > target_amount);
         assert_eq!(result.fee_amount, (50 + result.selected.len() * 68) as u64);
     }
+
+    #[test]
+    fn test_calculate_waste_with_change() {
+        let fee_rate = FeeRate::from_sat_per_vb(11.0);
+        let selected = generate_utxos_of_values(vec![100_000_000, 200_000_000]);
+        let amount_needed = 200_000_000;
+
+        let utxos: Vec<OutputGroup<'_>> = selected
+            .iter()
+            .map(|u| OutputGroup::new(u, fee_rate))
+            .collect();
+
+        let size_of_change = 31;
+        let cost_of_change: u64 = (size_of_change as f32 * fee_rate.as_sat_vb()) as u64;
```

Comment:
This may just be my limited proficiency in reading Rust, but above you described that `cost_of_change` should be composed of the cost of creating a change output and the cost of spending it later. This seems to only be composed of the cost of creating the change output, but I don't see the cost of spending it.

Reply:
So, using the [size-calulator](https://bitcoinops.org/en/tools/calc-size/) from optech, that same UTXO would weight 68 vbytes as a transaction input.
Am I missing something?

```rust
let change_out_size = 31;
let change_in_size = 68;
let cost_of_change: u64 = (
  change_out_size as f32 * fee_rate.as_sat_vb() - change_in_size as f32 * LONG_TERM_FEE_RATE.as_sat_vb()
) as u64;
```